### PR TITLE
Update go-ucfg

### PR DESCRIPTION
**Pull request by urso**
- Update go-ucfg
- add support for parsing lists/dictionaries from environment variables and via
  `-E` flag

### CODE REVIEW

**Comment by ruflin** [comment]
@urso Seems like some tests need an update.

**Comment by urso** [comment]
go-ucfg 0.4 adds support for parsing arrays and objects supporting syntax superset of JSON. Arrays of strings can be written by simply separating strings with commas.

Resolve: #2978

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -13,6 +13,8 @@ https://github.com/elastic/beats/compare/v5.0.0...master[Check the HEAD diff]
 ==== Breaking changes
 
 *Affecting all Beats*
+- Add support for passing list and dictionary settings via -E flag.
```
Comment (ruflin) [inline CHANGELOG.asciidoc]: What part of this change is breaking? It is an addition I think?

Comment (urso) [inline CHANGELOG.asciidoc]: hm... wrong section. Might have happened when rebasing

Comment (urso) [inline CHANGELOG.asciidoc]: Fixed.

**Comment by ruflin** [comment]
@urso CHANGELOG conflict :-( One more rebase needed.

**Comment by urso** [comment]
@ruflin done

**Comment by ApsOps** [comment]
@urso Hi. I just built from master, but still hitting this issue.

I use this config:
```yaml
filebeat.prospectors:
-
  paths:
    - "/var/log/containers/*.log"
  document_type: docker-logs

output.logstash:
    hosts: [${LOGSTASH_HOSTS:'logstash:5043'}]
    timeout: 15

logging.level: debug
```
and get error `Exiting: error loading config file: yaml: line 7: found unexpected ':'`

This works perfectly with filebeat v1.2.x

**Comment by ApsOps** [comment]
@urso Thanks!

I was testing this, built from master. It didn't work as expected, I thought maybe some edge case was missed. So commented here instead of forums 😄 

> Parsing has changed, env variables are not 'inserted' before parsing, but interpreted after parsing YAML => use quotes properly, like hosts: '${LOGSTASH_HOSTS:"logstash:5043"}'.

Is it possible to pass an array now? I'm not sure exactly what to use when setting env vars.
Using `LOGSTASH_HOSTS=["host1:5043","host2:5043"]` gives an error:
```
error initializing publisher: array expected ',' or ']' when parsing '[host1:5043,host2:5043]' accessing 'output.logstash.hosts'```

**Comment by urso** [comment]
1. As I said, you have to be careful about using quotes. As you can see from error message, your shell did remove the quotes.

go-ucfg supports a simplified syntax on top-level `LOGSTASH_HOSTS="host1:5043,host2:5043"`. The shell will remove the double quotes again, but without using `[]` it's ok.

2. Seriously, use discuss next time. Some other users have had same issue in past + new users with same issues might learn from it. That is, others but me can help as well.

**Comment by ApsOps** [comment]
@urso I'm sorry. I promise I'll be sure to use discuss in the future + will send a PR for this in docs!

Using `LOGSTASH_HOSTS="host1:5043,host2:5043"` still gives an error:
```
failed to initialize logstash plugin as output: can not convert 'object' into 'string' accessing 'output.logstash.hosts'
```
Using this in config:
```yaml
output.logstash:
    hosts: '${LOGSTASH_HOSTS:"logstash:5043"}'
    timeout: 15
```

**Comment by tsg** [comment]
This discussion was followed here: https://discuss.elastic.co/t/using-list-environment-variables-in-filebeat-config/70124/2

**Pull request merged**